### PR TITLE
142636481 refactor routes

### PR DESCRIPTION
**Pull request by drod1000**
@jdconrad89 @mollybrown @Laszlo-JFLMTCO 

I just finished refactoring the routes to make it a little cleaner. I updated the relevant tests and they're all passing except for the one that's failing due to the hot fix. Also, it seems like the current tests are treating projects as a single resource even though a user can have three. I didn't get too deep into it since I figured that's probably best left for another issue.

@neight-allen

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -9,18 +9,10 @@
...
-    put "portfolio", to: "users/portfolios#update"
-    post "portfolio", to: "users/portfolios#create"
-    delete "portfolio", to: "users/portfolios#destroy"
-    get "portfolio/delete", to: "users/portfolios#delete"
-    get "project/new", to: "users/projects#new"
-    post "project", to: "users/projects#create"
-    delete "project", to: "users/projects#destroy"
-    get "project/edit", to: "users/projects#edit"
-    put "project", to: "users/projects#update"
+    resource :portfolio, :controller => "users/portfolios", only: [:show, :new, :create, :edit, :update, :destroy] do
```
Comment (jdconrad89) [inline config/routes.rb]: @drod1000 This looks much better! Thanks for doing this!

**Comment by jdconrad89** [comment]
@drod1000 Thanks for refactoring this and for fixing the tests, looks much better!

File: config/routes.rb (hunk 1 above)
Comment (Laszlo-JFLMTCO) [inline config/routes.rb]: @drod1000 Looks great!!! I see that you were able to figure out to have the controller under `users/portfolios` instead of `portfolios`.

I think this line can further be refactored: `only: [:show, :new, :create, :edit, :update, :destroy]` is redundant as using `resource` (singular) already takes care of generating all routes except `index`. I attached screenshot confirming the routes are the same without `only: [:show, :new, :create, :edit, :update, :destroy]`.

Attached screenshot terminal window on the left is `rake routes` output of PR #84 code, terminal window on the right is `rake routes` output without `only: [:show, :new, :create, :edit, :update, :destroy]`

screen shot 2017-03-28 at 8 41 12 pm

So in conclusion I think this line should be the following:
`resource :portfolio, :controller => "users/portfolios" do`. With this shorter code still `rspec` output is the same as for PR #84.

@jdconrad89 @mollybrown What do you think?

FYI @neight-allen

File: config/routes.rb [hunk 2]
```diff
@@ -9,18 +9,10 @@
...
+      get '/delete', to: "users/portfolios#delete"
```
Comment (Laszlo-JFLMTCO) [inline config/routes.rb]: @drod1000 As we continue with the refactor effort as we discussed earlier today, let's look into this route again. `GET` verb with `delete` route seems a bit off...

FYI @jdconrad89 @mollybrown @neight-allen

**Comment by Laszlo-JFLMTCO** [comment]
@drod1000 Many thanks for pushing forward, merging latest `master` with `developer` branch and to take care of cleaning up the routes!!! I added a couple smart@ss comments for our review that could further clean up the `routes` file.

File: config/routes.rb (hunk 2 above)
Comment (jdconrad89) [reply config/routes.rb]: @Laszlo-JFLMTCO now that you mentioned that yea that does seem a bit off... I had looked at that line earlier and thought something was off but couldn't place it

File: config/routes.rb (hunk 1 above)
Comment (drod1000) [reply config/routes.rb]: @Laszlo-JFLMTCO I like it! Feel free to push it.

File: config/routes.rb (hunk 2 above)
Comment (drod1000) [reply config/routes.rb]: @Laszlo-JFLMTCO Yeah I still haven't figured out exactly what it's doing but the tests broke if I didn't include this. Definitely something we should look into.

File: config/routes.rb (hunk 1 above)
Comment (Laszlo-JFLMTCO) [reply config/routes.rb]: @drod1000 Done, I pushed it to this same branch: `142636481-refactor-routes`.

FYI @jdconrad89 @mollybrown

File: spec/features/portfolios/user_can_create_portfolio_spec.rb [hunk 3]
```diff
@@ -9,7 +9,7 @@
 
       click_link 'Create Profile'
 
-      expect(current_path).to eq(user_portfolio_new_path(user.id))
+      expect(current_path).to eq(new_user_portfolio_path(user.id))
```
Comment (mollybrown) [inline spec/features/portfolios/user_can_create_portfolio_spec.rb:12]: I like the improved path name.

File: config/routes.rb [hunk 4]
```diff
@@ -9,18 +9,10 @@
...
-    get "project/new", to: "users/projects#new"
-    post "project", to: "users/projects#create"
-    delete "project", to: "users/projects#destroy"
-    get "project/edit", to: "users/projects#edit"
-    put "project", to: "users/projects#update"
+    resource :portfolio, :controller => "users/portfolios" do
+      get '/delete', to: "users/portfolios#delete"
+    end
+    resource :project, :controller => "users/projects", only: [:new, :create, :edit, :update, :destroy]
   end
```
Comment (neight-allen) [inline config/routes.rb:16]: Much easier to read 👍  @drod1000
